app/permissions_migrations.go

```diff
@@ -268,32 +268,64 @@ func getViewMembersPermissionMigration() permissionsMap {
 			On:  isRole(model.SYSTEM_ADMIN_ROLE_ID),
 			Add: []string{PERMISSION_VIEW_MEMBERS},
 		},
-	}
+	}, nil
 }
 
-func getAddManageGuestsPermissionsMigration() permissionsMap {
+func getAddManageGuestsPermissionsMigration(a *App) (permissionsMap, error) {
 	return permissionsMap{
 		permissionTransformation{
 			On:  isRole(model.SYSTEM_ADMIN_ROLE_ID),
 			Add: []string{PERMISSION_PROMOTE_GUEST, PERMISSION_DEMOTE_TO_GUEST, PERMISSION_INVITE_GUEST},
 		},
-	}
+	}, nil
 }
 
-func getAddUseMentionChannelsPermissionMigration() permissionsMap {
-	return permissionsMap{
-		permissionTransformation{
+func getAddUseMentionChannelsPermissionMigration(a *App) (permissionsMap, error) {
+	transformations := permissionsMap{}
+
+	type rolePair struct {
+		targetRole string
+		otherRole  string
+	}
+
+	rolePairs := []rolePair{
+		{targetRole: model.CHANNEL_ADMIN_ROLE_ID, otherRole: model.CHANNEL_USER_ROLE_ID},
+	}
+
+	var allTeamSchemes []*model.Scheme
+
+	next := a.SchemesIterator(model.SCHEME_SCOPE_TEAM, permissionsExportBatchSize)
+	var schemeBatch []*model.Scheme
+	for schemeBatch = next(); len(schemeBatch) > 0; schemeBatch = next() {
+		allTeamSchemes = append(allTeamSchemes, schemeBatch...)
+	}
+
+	for _, ts := range allTeamSchemes {
+		rolePairs = append(rolePairs, rolePair{targetRole: ts.DefaultChannelAdminRole, otherRole: ts.DefaultChannelUserRole})
+	}
+
+	for _, rolePair := range rolePairs {
+		for perm := range model.ModeratedPermissions {
+			trans := permissionTransformation{
+				On:  permissionAnd(isRole(rolePair.targetRole), onOtherRole(rolePair.otherRole, permissionExists(perm))),
+				Add: []string{perm},
+			}
+			transformations = append(transformations, trans)
+		}
+		transformations = append(transformations, permissionTransformation{
```

Comment:
I feel like the additions to this migration should instead be in a new migration - since many people have already pulled changes from master and it would require them to delete the migration key and restart server in order to have the expected schema

Reply:
Also just so i understand is this adding any channel moderated permissions that exist on the channel user role onto the channel admin role?